docs/basic_construction.md

```diff
@@ -0,0 +1,155 @@
+
+### Tuple Construction via DFS
+- The prover constructs tuples recursively using DFS.
+- At each level $i$, the prover:
+  - Extends the current tuple by adding an element $s$ from $S_p$.
+  - Checks tuple validity using the random oracle $H_1$. A tuple $(t, s_1, \ldots, s_i)$ is valid if $H_1(t, s_1, \ldots, s_i) = 1$.
+  - If the tuple fails the $H_1$ check, it is discarded, and the DFS does not explore its descendants.
+  - Otherwise, the prover recursively calls DFS on the newly extended tuple to continue the construction process.
+- The process continues until a valid tuple of length $u$ is found, or all possibilities are exhausted.
+
+#### Example: how DFS works
+- Choose a root $t \in \[d\]$ and initialize the search with $(t)$.
+  - At each step, extend the current tuple $(t, s_1, \ldots, s_i)$ by trying every element $s \in S_p$.
+  - Check if $H_1(t, s_1, \ldots, s_i) = 1$:
+    - If true, recursively extend the tuple.
+    - If false, backtrack and try the next candidate for $s_i$.
+- Stop when a valid tuple $(t, s_1, \ldots, s_u)$ is found and satisfies $H_2(t, s_1, \ldots, s_u) = 1$.
+
+## The Protocol
+### Parameter Generation
+The parameter generation process for the basic construction is designed to ensure the protocol achieves the desired levels of *security* and *efficiency*.
+The internal parameters of ALBA, such as $u$, $d$, $q$, are computed based on the security parameters, $n_f$, $n_p$ .
+- Security parameters:
+  - $\lambda_{sec}: $ defines the probability that an adversary, who knows at most $n_f$ elements, can construct a valid proof. It controls the protocol's resistance to adversarial attacks by making forgery negligibly likely.
+  - $\lambda_{rel}: $ defines the probability that an honest prover, with knowledge of a sufficiently large set $S_p$, can successfully generate a valid proof. It ensures the practicality of the protocol for honest participants, emphasizing the importance of feasibility in proof construction.
+- Relationship between the parameters:
+  - The proof length $u$ is directly proportional to the security parameters and inversely proportional to the logarithm of the ratio $n_p / n_f$. 
+    - This means that as the prover's set size $n_p$ grows compared to the threshold $n_f$, the required proof length decreases logarithmically, reflecting the reduced adversarial advantage.
+      $$
+        u \leftarrow \Bigg\lceil \frac{\lambda_{sec} + \log \lambda_{rel} + 1 - \log \log e}{\log (n_p / n_f)}\Bigg\rceil.
+      $$
+  - The maximum number of subtrees to search $d$ depends linearly on both the proof length $u$ and the security parameter $\lambda_{rel}$. 
+    - It directly influences the ability to find a valid proof.
+    - Increasing $d$ improves the chances of constructing a valid proof, ensuring robustness in the presence of random oracle checks.
+      $$
+        d \leftarrow \Bigg\lceil \frac{2 u \lambda_{rel}}{\log e}\Bigg\rceil.
```

Comment:
nit: I think the left arrow is used for randomize computations. Maybe use := ? `\coloneqq` might work.

Reply:
You're right, it's a copy-paste leftover.